pkg/pillar/cmd/zedrouter/networkinstance.go

```diff
@@ -354,17 +354,44 @@ func doBridgeAclsDelete(
 			aclArgs := types.AppNetworkACLArgs{IsMgmt: false, BridgeName: ulStatus.Bridge,
 				VifName: ulStatus.Vif, BridgeIP: ulStatus.BridgeIPAddr, AppIP: ulStatus.AllocatedIPAddr,
 				UpLinks: status.IfNameList}
-			ruleList, err := deleteACLConfiglet(aclArgs, ulStatus.ACLRules)
+			rules := getNetworkACLRules(ctx, appID, ulStatus.Name)
+			ruleList, err := deleteACLConfiglet(aclArgs, rules.ACLRules)
 			if err != nil {
 				log.Errorf("NetworkInstance DeleteACL failed: %s\n",
 					err)
 			}
-			ulStatus.ACLRules = ruleList
+			setNetworkACLRules(ctx, appID, ulStatus.Name, ruleList)
 		}
 	}
 	return
 }
 
+func getNetworkACLRules(ctx *zedrouterContext, appID uuid.UUID, intf string) types.ULNetworkACLs {
+	tmpMap := ctx.NLaclMap[appID]
+	if tmpMap == nil {
+		ctx.NLaclMap[appID] = make(map[string]types.ULNetworkACLs)
+	}
+
+	if _, ok := ctx.NLaclMap[appID][intf]; !ok {
+		ctx.NLaclMap[appID][intf] = types.ULNetworkACLs{}
+	}
+	return ctx.NLaclMap[appID][intf]
+}
+
+func setNetworkACLRules(ctx *zedrouterContext, appID uuid.UUID, intf string, rulelist types.IPTablesRuleList) {
+	tmpMap := ctx.NLaclMap[appID]
+	if tmpMap == nil {
+		ctx.NLaclMap[appID] = make(map[string]types.ULNetworkACLs)
+	}
+
+	if len(rulelist) == 0 {
+		delete(ctx.NLaclMap[appID], intf)
```

Comment:
Don't we need some code to delete(ctx.NLaclMap, appID) somewhere when the appID goes away?

Reply:
this is in 
@@ -1443,6 +1448,8 @@ func appNetworkDoInactivateAllUnderlayNetworks(
                appNetworkDoInactivateUnderlayNetwork(
                        ctx, status, ulStatus, ipsets)
        }
+       appID := status.UUIDandVersion.UUID
+       delete(ctx.NLaclMap, appID)

with so many yetus errors, it's hard to notice.

Reply:
Found that code.